fv3config/config/nudging.py

```diff
@@ -0,0 +1,144 @@
+        if hour <= current_hour:
+            first_nudge_hour = hour
+    return first_nudge_hour
+
+
+def _get_nudge_time_list(run_duration, current_date) -> List[datetime]:
+    """Return list of datetime objects corresponding to times at which analysis files
+    are required for nudging for a given model run configuration"""
+    start_time = datetime(*current_date)
+    first_nudge_time = _most_recent_nudge_time(start_time)
+    nudge_duration = run_duration + (start_time - first_nudge_time)
+    nudge_duration_hours = int(
+        math.ceil(nudge_duration.total_seconds() / SECONDS_IN_HOUR)
+    )
+    nudge_interval = NUDGE_HOURS[1] - NUDGE_HOURS[0]
+    nudging_hours = range(0, nudge_duration_hours + nudge_interval, nudge_interval)
+    return [first_nudge_time + timedelta(hours=hour) for hour in nudging_hours]
+
+
+def get_nudging_assets(
+    run_duration: timedelta,
+    current_date: Sequence[int],
+    nudge_url: str,
+    nudge_filename_pattern: str = "%Y%m%d_%HZ_T85LR.nc",
+    copy_method: str = "copy",
+) -> List[Mapping]:
+    """Return list of assets of nudging files required for given run duration and
+    start time.
+    
+    Args:
+        run_duration: length of fv3gfs run
+        current_date: start time of fv3gfs run as a sequence of 6 integers
+        nudge_url: local or remote path to nudging files
+        nudge_filename_pattern: template for nudging filenames. Defaults to
+            '%Y%m%d_%HZ_T85LR.nc'.
+        copy_method: copy_method for nudging file assets. Defaults to 'copy'.
+
+    Returns:
+        list of all assets required for nudging run
+
+    Raises:
+        ConfigError: if copy_method is "link" and a remote path is given for nudge_url
+    """
+    if get_fs(nudge_url) != fsspec.filesystem("file") and copy_method == "link":
+        raise ConfigError(
+            "Cannot link nudging files if using remote path for nudge_url. "
+            f"Got {nudge_url}."
+        )
+    time_list = _get_nudge_time_list(run_duration, current_date)
+    filename_list = [time.strftime(nudge_filename_pattern) for time in time_list]
+    nudging_assets = [
+        get_asset_dict(
+            nudge_url, file_, target_location="INPUT", copy_method=copy_method
+        )
+        for file_ in filename_list
+    ]
+    return nudging_assets
+
+
+def _non_nudging_assets(
+    assets: Sequence[Mapping], nudge_filename_pattern: str,
+) -> List[Mapping]:
+    """Given list of assets, return filtered list with no nudging assets.
+
+    Args:
+        assets: sequence of assets
+        nudge_filename_pattern: datetime pattern of nudging files. Defaults to
+            '%Y%m%d_%HZ_T85LR.nc'.
+        input_list_filename: name of text file listing all nudging files. Defaults to
+            'nudging_file_list'.
+    """
+    is_nudging_asset = partial(_target_name_matches, pattern=nudge_filename_pattern)
+    return [item for item in assets if not is_nudging_asset(item)]
+
+
+def _target_name_matches(asset, pattern):
```

Comment:
nit: maybe call this `_is_nudging_asset`?  I realize you use a similar name for the partial function above, but I think `_target_name_matches` is a bit vague given its somewhat nudging-file-specific behavior (i.e. checking that the target_name can be parsed for a date with a given time pattern).

Reply:
I went back and forth on this, but going to leave it as is, since the pattern is an argument to the function, so it doesn't really know anything about nudging per se.